onnxruntime/core/providers/cpu/tensor/cast_op.h

```diff
@@ -94,6 +131,20 @@ class Cast final : public OpKernel {
     return Status::OK();
   }
 
+  template <typename SrcType>
+  Status CastToStringData(const Tensor* in, Tensor* out, const TensorShape& shape) const {
+    ::onnxruntime::CastToStringData<SrcType>(in, out, shape);
+    return Status::OK();
+  }
+
+  template <typename DstType>
+  Status CastFromStringData(const Tensor* in, Tensor* out, const TensorShape& shape, OpKernelContext* context) const {
+    AllocatorPtr allocator;
+    ORT_RETURN_IF_ERROR(context->GetTempSpaceAllocator(&allocator));
+    ::onnxruntime::CastFromStringData<DstType>(in, out, shape, allocator);
+    return Status::OK();
+  }
+
   ONNX_NAMESPACE::TensorProto_DataType to_;
```

Comment:
Why do we need separate logic in ::onnxruntime::CastToStringData and ::onnxruntime::CastFromStringData? Can we combine the to and from into single functions?

Reply:
I tried many ways to combine them into one, but if we have any if-condition logic, the compiler will still compile the part when there is no method suitable for it and cause compile error. 

One example is:
```
if (std::is_same<DstType, std::string>){
}else{
  CastData<double, DstType>(&tmp_tensor, out, shape);
}
```
will cause compile error, for we cannot use static_cast to cast a double into string. That how i ended implemented separated functions for casting to/from string.